src/psyclone/psyir/transformations/intrinsics/mms_base_trans.py

```diff
@@ -203,170 +187,97 @@ def apply(self, node, options=None):
-            # There is no bounds information in the array reference,
-            # so look at the declaration.
-            # Note, the potential 'if not array_ref.symbol.is_array:'
-            # exception is already handled by the validate method.
-            ndims = len(array_ref.datatype.shape)
-
-            loop_bounds = []
-            for idx, shape in enumerate(array_ref.datatype.shape):
-                if shape in [ArrayType.Extent.DEFERRED,
-                             ArrayType.Extent.ATTRIBUTE]:
-                    if shape == ArrayType.Extent.DEFERRED:
-                        allocatable = True
-                    # runtime extent using LBOUND and UBOUND required
-                    lbound = IntrinsicCall.create(
-                        IntrinsicCall.Intrinsic.LBOUND,
-                        [Reference(array_ref.symbol),
-                         ("dim", Literal(str(idx+1), INTEGER_TYPE))])
-                    ubound = IntrinsicCall.create(
-                        IntrinsicCall.Intrinsic.UBOUND,
-                        [Reference(array_ref.symbol),
-                         ("dim", Literal(str(idx+1), INTEGER_TYPE))])
-                    loop_bounds.append((lbound, ubound))
-                elif isinstance(shape, ArrayType.ArrayBounds):
-                    # array extent is defined in the array declaration
-                    loop_bounds.append(shape)
-                #  Note, the validate method guarantees that an else
-                #  clause is not required.
-        else:
-            # The validate method guarantees that this is an array
-            # reference.
-            loop_bounds = []
-            ndims = len(array_ref.children)
-            for shape in array_ref.children:
-                loop_bounds.append((shape.start.copy(), shape.stop.copy()))
-
-        # Determine the datatype of the array's values and create a
-        # scalar of that type
-        array_intrinsic = array_ref.datatype.intrinsic
-        array_precision = array_ref.datatype.precision
-        scalar_type = ScalarType(array_intrinsic, array_precision)
-
-        symbol_table = node.scope.symbol_table
-        assignment = node.ancestor(Assignment)
-
-        datatype = scalar_type
-        array_reduction = False
-        if dimension_ref and ndims > 1:
-            array_reduction = True
-            # We are reducing from one array to another
-            shape = []
-            for idx, bounds in enumerate(loop_bounds):
-                # The validation constrains the transformation to only
-                # allow cases where the literal value for dimension
-                # is known.
-                if int(dimension_literal.value)-1 == idx:
-                    # This is the dimension we are performing the
-                    # reduction over so do not loop over it.
+        expr, _, mask_ref = self._get_args(node)
+
+        # Step 1 extract the expression within the intrinsic and put
```

Comment:
I think each Step comment would benefit with an example comment line similar to https://github.com/stfc/PSyclone/issues/2323#issuecomment-1784897338
otherwise is a bit difficult to follow each step.

Reply:
OK, I've hopefully done that to your satisfaction.